lib/server.js

```diff
@@ -255,6 +255,9 @@ function expiry(block2cache, k) {
 
 CoAPServer.prototype._handle = function(packet, rsinfo) {
 
+  // Make sure this is a request
+  if (packet.code[0] != '0') return;
+
```

Comment:
Ok. Could you add a bigger comment explaining this is needed to reuse the socket?

The problem with this change is when the socket is not reused, as the spec says we need to reply with something (couldn't remember now, please check), if we couldn't recognize the message.